### DISCUSSION TITLE
使用 HardHat 创建 NFT 智能合约 - forecho's Blog

### DISCUSSION THREAD

**Discussion by giscus** [forecho/hugo-blog#43] [Announcements]
# 使用 HardHat 创建 NFT 智能合约 - forecho's Blog

引言
最近在学智能合约，本篇文章主要记录如何使用 HardHat 创建 NFT 智能合约。我们还将学习如何使用 Hardhat 写测试合约和部署智能合约。
先决条件

略懂 Node.js
略懂 Solidity
略懂什么是区块链钱包，比如 MetaMask

搭建环境
安装 Node.js
首先，我们需要安装 Node.js，参照官网的安装教程即可。
MetaMask

安装并下载 MetaMask 浏览器插件，然后创建一个钱包。用于与区块链交互。
安装完 MetaMask 后，添加与 Polygon Mumbai 测试网的连接。另外也可以使用 Chainlist 快捷添加。
使用 Polygon 水龙头为你的账户充值测试网的 MATIC 代币。

https://blog.forecho.com/create-nft-smart-contract-with-hardhat.html

**Comment by shibaijin** [comment]
_exists()函数是否在新的ERC721中弃用了，跑合约的时候显示Undeclared identifier，浏览过新的ERC721包代码里也没有_exists()函数
